### ISSUE TITLE
Can't Use NPM 'request' Module

### ISSUE THREAD

**Issue by micahasmith** [Azure/azure-functions-host#2074]
#### Investigative information

Please provide the following:

- Timestamp: now
- Function App version (1.0 or 2.0-beta): ?
- Function App name: goot-fn-nodejs
- Function name(s) (as appropriate): test
- Invocation ID: `2017-10-27T14:15:58.246 Function started (Id=1322e92f-e683-4ca1-a508-b3c30fc710fd)`
- Region: East US

#### Repro steps

Provide the steps required to reproduce the problem:

Make a file with the following:

```js
var rq = require('request');

module.exports = function (context, req) {
    context.log('JavaScript HTTP trigger function processed a request.');

    if (req.query.name || (req.body && req.body.name)) {
        context.res = {
            // status: 200, /* Defaults to 200 */
            body: "Hello " + (req.query.name || req.body.name)
        };
    }
    else {
        context.res = {
            status: 400,
            body: "Please pass a name on the query string or in the request body"
        };
    }
    context.done();
};
```

#### Expected behavior

I would expect the request module to be importable, as its incredibly popular and used in many nodejs modules.

#### Actual behavior

Any nodejs file that uses it causes the function to run without end, never executing the core code.

#### Known workarounds

Move to another serverless provider. jks

#### Related information 

Provide any related information 

```js
{
  "name": "goot-bot",
  "version": "1.0.0",
  "description": "goot bot",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "async": "^2.5.0",
    "request": "2.34.1",
    "sequelize": "^4.18.0"
  }
}
```

**Comment by micahasmith** [comment]
is this some sort of trick to try to get me to move to fsharp

**Comment by christopheranderson** [comment]
Did you import the request module? (Bring your own node_module, npm i request, package.json and gitpush?)

**Comment by christopheranderson** [comment]
Just a package.json won't work by itself, needs the explicit import step or gitpush which does it for you. Also, did you have an actual error/stack trace? Can't see what might be wrong.

**Comment by pragnagopa** [comment]
@mhoeger - can you please follow up?

**Comment by mhoeger** [comment]
Hey @micahasmith, any updates on this issue?

**Comment by theonemule** [comment]
The built in HTTPS and HTTP modules seem to do the same thing. I tried these in response to request not working, so it's more fundamental than request.

**Comment by mhoeger** [comment]
@theonemule huh, I'm not able to repro this issue! This is the code that I'm able to run. Can you give me an example of your failing function?

Could it be related to this? We're still investigating, but I wonder if this is related to [Azure Web App Sandbox restrictions](https://github.com/projectkudu/kudu/wiki/Azure-Web-App-sandbox) as pragnagopa suggests (see her comment: https://github.com/Azure/azure-functions-nodejs-worker/issues/68)

```javascript
var https = require("https");
module.exports = function (context, req) {
    context.log("started");
    https.get("https://raw.githubusercontent.com/Azure/azure-functions-host/dev/README.md", (resp) => {
        var data = "";
        resp.on("data", (chunk) => {
            data += chunk;
            });
        resp.on("end", () => {
            context.log(data);
            context.done();
        });
    });
};
```

**Comment by theonemule** [comment]
@mhoeger

Thanks... I think it has to do with the limitations. It seems that when I have multiple nested callbacks such as calling out to Azure Storage (which uses request) and or other HTTP/HTTPS endpoints more than a few times in a function, then it fails. It works for one-offs like this seem to work.

**Comment by mhoeger** [comment]
@theonemule if it's a Sandbox restriction, it would always fail, but you're saying it works more than a few times? Could you share a code sample? I'd love to know what's going on!

**Comment by theonemule** [comment]
Sure...

This function is called within the function app I was trying to write. It's using Azure Table Storage to read and write data. I seems to die after it makes the initial call to Table Storage. I dug into the storage source code and found that it was using request to do HTTP calls too.


 
````
function auth(context, req, tableSvc){

    var parseQuery = function (queryString) {
        var query = {};
        var pairs = (queryString[0] === '?' ? queryString.substr(1) : queryString).split('&');
        for (var i = 0; i < pairs.length; i++) {
            var pair = pairs[i].split('=');
            query[decodeURIComponent(pair[0])] = decodeURIComponent(pair[1] || '');
        }
        return query;
    };    
    
    var data = parseQuery(req.body);
    var aad = new azurejwt(context) ;
    aad.verify(data.id_token, function(err, data){
        if(err){
			sendError(context, err);
			return;
        }else{
            
            context.log(data);
            
            var cuuid = data.nonce;
    	    tableSvc.createTableIfNotExists('sessions', function(error, result, response){
    			if(error){
    				redirectAuth(context, req);
    				return;
    			}
    			
        		var session = {
        			PartitionKey: {'_':'session'},
        			RowKey: {'_': cuuid},
        			Email: data.email
        		};    			
    			
        		tableSvc.insertEntity('session',share, function (tblerror, result, response) {
        			if(tblerror){
        				sendError(context, tblerror);
        				return;
        			}
                	var file="index.html";
                		
                	fs.readFile(__dirname + "/" +  file, "utf8", function(err, data) {
                		if (err){
            				sendError(context, tblerror);
            				return;                		    
                		}

            			var contentType = mime.lookup(file) ;
            			
            			context.res = {
            				status: 200, 
            				body: data,
            				isRaw: true,
            				headers: {
            					'Content-Type': contentType,
            					'Set-Cookie': "cuuid=" + cuuid
            				}
            			};
            			context.done();return;
                	});	
        		});    			
            });
        }
    })
}


````

**Comment by mhoeger** [comment]
Fixed with request "^2.88.0" and functions runtime version 2.0.12134.
